functions/grapher/[slug].ts

```diff
@@ -5,17 +5,30 @@ export const onRequestGet: PagesFunction = async (context) => {
 
     const { request, env, params } = context
 
-    const slug = params.slug as string
+    const originalSlug = params.slug as string
     const url = new URL(request.url)
 
     // All our grapher slugs are lowercase by convention.
     // To allow incoming links that may contain uppercase characters to work, we redirect to the lowercase version.
+    // We also then resolve redirects using the /grapher/_grapherRedirects.json file.
     // We do this before we even know that the page exists, so even if we redirect it might still be a 404, or go into another redirect.
-    if (
-        url.pathname !== url.pathname.toLowerCase() &&
-        url.pathname !== "/grapher/embedCharts.js"
-    ) {
-        const redirUrl = url.pathname.toLowerCase() + url.search
+    let currentSlug = originalSlug.toLowerCase()
+    const redirects = await env.ASSETS.fetch(
+        new URL("/grapher/_grapherRedirects.json", url),
+        { cf: { cacheTtl: 2 * 60 } }
+    )
```

Comment:
This is really my biggest question in working on this, what is the best way to efficiently fetch the redirects file, which we're gonna have to await on every request now.
Other options include:
- Storing redirects in Workers KV, somehow
- Uploading the redirects.json as part of the worker (ugh, but that's probably actually the fastest way)

I'm also not entirely sure whether the CF caching header will actually do something meaningful here, but it probably can't hurt also, and maybe it can actually speed things up?

Reply:
I was surprised by how fast is (uncached) fetch from R2, so I'd say that fetching redirects from assets explicitly should be light speed. It could be worth logging latency just to be sure.